src/server/replica.cc

```diff
@@ -704,6 +704,18 @@ error_code Replica::ConsumeRedisStream() {
       ThisFiber::Yield();
     }
 
+    // If the acks-fb or something else triggered a shutdown, then do not attempt to read from the
+    // stream.
+    if (!exec_st_.IsRunning()) {
+      if (exec_st_.IsError()) {
+        LOG_REPL_ERROR("Stopping stream consumer in phase "
+                       << GetCurrentPhase()
+                       << " because of external error: " << exec_st_.GetError().Format());
+      }
+      acks_fb_.JoinIfNeeded();
+      return exec_st_.GetError();
```

Comment:
Is it possible we end up here without an error? I can't see any code path where that could happen but I don't think adding a `DCHECK` here is sensible

Reply:
I see. You were not talking about in general, if the error can be empty (which it can if it got cancelled via `Cancel()`). 

In `replica.cc` context -- you are right, we always cancel via `ReportCancelError` so there should be an error always.

In that case I suggest to: (see the code block)